app/helpers/application_helper.rb

```diff
@@ -133,52 +137,72 @@ def link_prev(object)
   #     name: :destroy_object.t(type: :herbarium),
   #     target: herbarium_path(@herbarium, back: url_after_delete(@herbarium))
   #   )
-  def destroy_button(target:, name: :DESTROY.t)
+  def destroy_button(target:, name: :DESTROY.t, **args)
     path = if target.is_a?(String)
              target
            else
              add_query_param(send("#{target.type_tag}_path", target.id))
            end
-    id = target.is_a?(String) ? nil : "destroy_#{target.type_tag}"
+    classes ||= "text-danger"
+    id ||= nil
+    unless target.is_a?(String)
+      classes += " destroy_#{target.type_tag}_link_#{target.id}"
+    end
 
-    button_to(
-      name, path, method: :delete, class: "text-danger", id: id,
-                  data: { confirm: :are_you_sure.t }
-    )
+    html_options = {
+      method: :delete,
+      class: classes,
+      id: id,
+      data: { confirm: :are_you_sure.t }
+    }.merge(args)
+
+    button_to(name, path, html_options)
   end
 
   # POST to a path; used instead of a link because POST link requires js
   # post_button(name: herbarium.name.t,
   #             path: herbaria_merges_path(that: @merge.id,this: herbarium.id),
-  #             confirm: :are_you_sure.t)
-  def post_button(name:, path:, confirm: nil)
-    data = confirm ? { confirm: confirm } : nil
-    button_to(name, path, method: :post, class: "text-info", data: data)
+  #             data: { confirm: :are_you_sure.t })
+  def post_button(name:, path:, **args)
+    html_options = {
+      method: :post,
+      class: "text-info"
```

Comment:
This is such a minor thing, and totally not important, but I wonder if maybe you have seen this problem solved in your perusal of forums and such...  In an ideal world, this would not _replace_ the CSS class, but add to it, right?  I bet this is a common thing.  Basically where you want a merge but where it concatenates the new value if it's not already in there.  I could almost see it being worthwhile to write little Hash extensions which do this:

html_args = args.overwrite(method: :post).concat(class: "text-info")

So totally not important here because of course none of your callers have provided the "class" argument anyway! :)  But just thought I'd throw that out there in case it resonated as something worth doing some day...

Reply:
YES! I was puzzling over this. Thank you for raising it.
Sometimes, though, I do want to replace the class, not concatenate. So I left it like this for now. "text-info" in Bootstrap enforces a text color, and I figured i might want to change the color in some places. Or make it a button, or whatever.

Reply:
Exactly, my suggestion would let you choose how to treat each argument, cleanly, transparently, and conveniently.  (I always thought the args = { override: this }.merge(input_args) mechanism was clunky.)